### PR TITLE
Filter jobs that are over quota out of /queue endpoint

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Filter jobs out of the /queue endpoint that will exceed the user's quota

## Why are we making these changes?
This prevents users that submit a large number of jobs outside their quota from clogging up the /queue output.

### CODE REVIEW

**Comment by dposada** [comment]
Build is red

**Comment by pschorf** [comment]
Green again :) ✔️

File: integration/tests/cook/test_multi_user.py [hunk 1]
```diff
@@ -485,3 +485,55 @@ def test_user_total_usage(self):
...
+        default_pool = util.default_pool(self.cook_url)
+        pool = default_pool or 'no-pool'
+        def queue_uuids():
+            try:
+                queue = util.query_queue(self.cook_url).json()
+                uuids = [j['job/uuid'] for j in queue[pool] if j['job/user'] == user.name]
+                self.logger.info(f'Queued uuids: {uuids}')
+                return uuids
+            except BaseException as e:
+                self.logger.error(f"Didn't reach desired instance count: {e}")
```
Comment (dposada) [inline integration/tests/cook/test_multi_user.py]: I don't understand the error message here

File: scheduler/test/cook/test/mesos/util.clj [hunk 2]
```diff
@@ -695,4 +695,140 @@
...
+      (is (= []
+             (util/filter-based-on-quota {test-user {:count 1, :cpus 2, :mem 1024}} user->usage queue))))
+    (testing "all jobs included"
+      (is (= [(make-job 1 2 2048) (make-job 2 1 1024) (make-job 3 3 4096) (make-job 4 1 1024)]
+             (util/filter-based-on-quota {test-user {:count 10, :cpus 20, :mem 32768}} user->usage queue))))
+    (testing "room for later jobs not included"
+      (is (= [(make-job 1 2 2048) (make-job 2 1 1024)]
+             (util/filter-based-on-quota {test-user {:count 4, :cpus 20, :mem 6144}} user->usage queue))))))
+
+
```
Comment (dposada) [inline scheduler/test/cook/test/mesos/util.clj]: Extra newline?

File: integration/tests/cook/test_multi_user.py (hunk 1 above)
Comment (pschorf) [reply integration/tests/cook/test_multi_user.py]: Bad copy paste :(

File: scheduler/test/cook/test/mesos/util.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/test/cook/test/mesos/util.clj]: Done

**Pull request merged**